core/src/main/java/com/linecorp/armeria/common/stream/ByteStreamMessageOutputStream.java

```diff
@@ -0,0 +1,238 @@
+    public boolean isOpen() {
+        return delegate.isOpen();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return delegate.isEmpty();
+    }
+
+    @Override
+    public long demand() {
+        return delegate.demand();
+    }
+
+    @Override
+    public CompletableFuture<Void> whenComplete() {
+        return delegate.whenComplete();
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super HttpData> subscriber, EventExecutor eventExecutor,
+                          SubscriptionOption... options) {
+        final Subscriber<HttpData> outputStreamSubscriber = new OutputStreamSubscriber(
+                subscriber, eventExecutor, streamWriter, outputStreamWriter, executor);
+        delegate.subscribe(outputStreamSubscriber, eventExecutor, options);
+    }
+
+    @Override
+    public void abort() {
+        delegate.abort();
+    }
+
+    @Override
+    public void abort(Throwable cause) {
+        delegate.abort(cause);
+    }
+
+    private static final class OutputStreamSubscriber implements Subscriber<HttpData>, Subscription {
+
+        private final Subscriber<? super HttpData> downstream;
+        private final EventExecutor eventExecutor;
+
+        private final StreamMessageAndWriter<HttpData> streamWriter;
+        private final Consumer<OutputStream> outputStreamWriter;
+        private final Executor executor;
+
+        @Nullable
+        private Subscription upstream;
+        private boolean completed;
+
+        OutputStreamSubscriber(Subscriber<? super HttpData> downstream,
+                               EventExecutor eventExecutor,
+                               StreamMessageAndWriter<HttpData> streamWriter,
+                               Consumer<OutputStream> outputStreamWriter,
+                               Executor executor) {
+            requireNonNull(downstream, "downstream");
+            requireNonNull(eventExecutor, "eventExecutor");
+            requireNonNull(streamWriter, "streamWriter");
+            requireNonNull(outputStreamWriter, "outputStreamWriter");
+            requireNonNull(executor, "executor");
+            this.downstream = downstream;
+            this.eventExecutor = eventExecutor;
+            this.streamWriter = streamWriter;
+            this.outputStreamWriter = outputStreamWriter;
+            this.executor = executor;
+        }
+
+        @Override
+        public void onSubscribe(Subscription subscription) {
+            requireNonNull(subscription, "subscription");
+            upstream = subscription;
+            downstream.onSubscribe(this);
+            CompletableFuture.runAsync(() -> outputStreamWriter
+                                     .accept(new StreamWriterOutputStream(streamWriter)), executor)
+                             .whenComplete((res, cause) -> {
+                                 if (cause != null) {
+                                     onError(cause);
+                                 }
+                             });
```

Comment:
To propagate error inside of `outputSTreamWriter.accept(..)` to downStream well!
Without this, `@Test write_error_thrown()` test not finished cause it can't get `onError` from executor well.

If you have any idea then feel free to share to me! 😄